### PR TITLE
python311Packages.abjad: fix build

### PR DESCRIPTION
**Pull request by davisrichard437**
## Description of changes

Two changes:

1. The `abjad` build was broken with a ModuleNotFoundError, so I added the relative module to dependencies.
2. The `--replace` option was deprecated, so I replaced it with `--replace-fail` for robustness.

Note: I seem to be unable to run `nixpkgs_review`; the `nix-env -qaP` command (with some extra flags) required for the script seems to fail and I'm not sure why.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by jtbx — APPROVED** [review]
> Note: I seem to be unable to run `nixpkgs_review`; the `nix-env -qaP` command (with some extra flags) required for the script seems to fail and I'm not sure why.

It might have been killed by the OOM killer, I fixed it by allocating a few gigs of swap space. nixpkgs-review does seem to take up quite a lot of memory.

LGTM :)

**Comment by davisrichard437** [comment]
Hello all, not sure what the process is for this, but how do I go about pinging people for review? I realize now this has been open for more than a month and I'd like to get it merged if possible

**Comment by jtbx** [comment]
Normally you would ask on the NixOS discourse, on a thread such as [PRs ready for review](https://discourse.nixos.org/t/prs-ready-for-review/3032). That will usually bring your PR to attention.

**Comment by jtbx** [comment]
Hey I just realised your branch has conflicts with the master branch, meaning it can't be merged into it cleanly. Rebase against nixpkgs master and solve the conflicts from there. Force-push to your PR branch and you should be good to go.

**Comment by davisrichard437** [comment]
Done I think!

File: pkgs/development/python-modules/abjad/default.nix [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
+{ lib
+, buildPythonPackage
+, fetchPypi
+, ply
+, roman
+, uqbar
+, pythonOlder
+, pytestCheckHook
+, lilypond
+, typing-extensions
```
Comment (jtbx) [inline pkgs/development/python-modules/abjad/default.nix:10]: Can you please remove this reformatting, it's unnecessary

Comment (davisrichard437) [reply pkgs/development/python-modules/abjad/default.nix:10]: Even if it's specified that way in the [code conventions](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#syntax) section of the contributing page?

Comment (jtbx) [reply pkgs/development/python-modules/abjad/default.nix:10]: Oh I didn't notice that bit, if you want to keep it please put the change in another commit. Also please change the PR title to something like `abjad: fix build`.

Comment (davisrichard437) [reply pkgs/development/python-modules/abjad/default.nix:10]: All done!

**Review by jtbx — APPROVED** [review]
Awesome thanks for all that. :+1:

File: pkgs/development/python-modules/abjad/default.nix (hunk 1 above)
Comment (natsukium) [reply pkgs/development/python-modules/abjad/default.nix:10]: Could you please revert this change?
We have adopted `nixfmt-rfc-style` formatting for python modules. https://github.com/NixOS/nixpkgs/pull/313628

**Comment by natsukium** [comment]
Since this package is a python library, please write the commit messages and PR title starting with `python311Packages.abjad`.

**Comment by davisrichard437** [comment]
Done!

File: pkgs/development/python-modules/abjad/default.nix (hunk 1 above)
Comment (davisrichard437) [reply pkgs/development/python-modules/abjad/default.nix]: @natsukium reverted!

**Review by natsukium — APPROVED** [review]
LGTM, thanks!

**Pull request merged**